pkg/sql/schemachanger/scbuild/internal/scbuildstmt/process.go

```diff
@@ -11,31 +11,85 @@
 package scbuildstmt
 
 import (
+	"reflect"
+
 	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scerrors"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessiondatapb"
+	"github.com/cockroachdb/errors"
 )
 
+// supportedStatement tracks metadata for statements that are
+// implemented by the new schema changer.
+type supportedStatement struct {
+	fn             interface{}
+	fullySupported bool
+}
+
+// IsFullySupported returns if this statement type is supported, where the
+// mode of the new schema changer can force unsupported statements to be
+// supproted.
+func (s supportedStatement) IsFullySupported(mode sessiondatapb.NewSchemaChangerMode) bool {
+	// If the unsafe modes of the new schema changer are used then any implemented
+	// operation will be exposed.
+	if mode == sessiondatapb.UseNewSchemaChangerUnsafeAlways ||
+		mode == sessiondatapb.UseNewSchemaChangerUnsafe {
+		return true
+	}
+	return s.fullySupported
+}
+
+// Tracks operations which are fully supported when the declarative schema
+// changer is enabled. Operations marked as non-fully supported can only be
+// with the experimental_use_new_schema_changer session variable.
+var supportedStatements = map[reflect.Type]supportedStatement{
+	// Alter is fully supported, but subcommands may not be.
```

Comment:
This seems like a contradiction in terms, to me at least: if it's fully supported, then I'd expect that to mean all subcommands would also be.